src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        internal static IReadOnlyCollection<IOTelExtension> TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            var targetFramework = FrameworkDescription.Instance.TargetFramework;
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins with target framework '{0}'.", targetFramework);
+
+            // TODO: Detect if objects instead of path
+            var pluginFiles = pluginsConfig.GetValue<JToken>($"['{targetFramework}']").ToObject<string[]>();
```

Comment:
I think it will throw an exception in case of case of bad JSON structure.
https://github.com/JamesNK/Newtonsoft.Json/blob/f7e7bd05d9280f17993500085202ff4ea150564a/Src/Newtonsoft.Json/Linq/JToken.cs#L1961
There would be lo log message in such a case.

Reply:
Seems right, fixed.